File: backend/src/main/kotlin/ch/yvu/teststore/result/TestWithResults.kt
package ch.yvu.teststore.result

data class TestWithResults(val testName: String, val results: List<Result> = emptyList()) {
    enum class TestResult {
        PASSED, FAILED, RETRIED
    }

    fun addResult(result: Result): TestWithResults {
        return TestWithResults(testName, results + result)
    }

    fun getTestResult(): TestResult {
        if (results.isEmpty()) return TestResult.PASSED
        val lastResult = results.sortedBy { it.retryNum }.last()

        if (!lastResult.isPassed()) return TestResult.FAILED
        else if (lastResult.isRetry()) return TestResult.RETRIED
        else return TestResult.PASSED
    }
}

File: web/www/app/test-result/test-with-results.ts
import {TestResult} from "./test-result";

export class TestWithResults {
    testName: string;
    results: TestResult[];
    testResult: string;

    getRunId(): string {
        if(this.results == null || this.results.length == 0) {
            return null;
        }

        return this.results[0].run;
    }
}